pkg/repository/prisma/tenant.go

```diff
@@ -169,6 +179,82 @@ func (r *tenantAPIRepository) DeleteTenantMember(memberId string) (*db.TenantMem
 	).Delete().Exec(context.Background())
 }
 
+func (r *tenantAPIRepository) GetQueueMetrics(tenantId string, opts *repository.GetQueueMetricsOpts) (*repository.GetQueueMetricsResponse, error) {
+	if err := r.v.Validate(opts); err != nil {
+		return nil, err
+	}
+
+	totalParams := dbsqlc.GetTenantTotalQueueMetricsParams{
+		TenantId: sqlchelpers.UUIDFromStr(tenantId),
+	}
+
+	workflowParams := dbsqlc.GetTenantWorkflowQueueMetricsParams{
+		TenantId: sqlchelpers.UUIDFromStr(tenantId),
+	}
+
+	if opts.AdditionalMetadata != nil {
+		additionalMetadataBytes, err := json.Marshal(opts.AdditionalMetadata)
+		if err != nil {
+			return nil, err
+		}
+
+		totalParams.AdditionalMetadata = additionalMetadataBytes
+		workflowParams.AdditionalMetadata = additionalMetadataBytes
+	}
+
+	if opts.WorkflowIds != nil {
+		uuids := make([]pgtype.UUID, len(opts.WorkflowIds))
+
+		for i, id := range opts.WorkflowIds {
+			uuids[i] = sqlchelpers.UUIDFromStr(id)
+		}
+
+		workflowParams.WorkflowIds = uuids
+		totalParams.WorkflowIds = uuids
+	}
+
+	tx, err := r.pool.Begin(context.Background())
+
+	if err != nil {
+		return nil, err
+	}
+
+	defer deferRollback(context.Background(), r.l, tx.Rollback)
```

Comment:
why are we deferring a rollback for read only queries?

Reply:
doesn't seem to make a difference: https://www.postgresql.org/message-id/flat/07FDEE0ED7455A48AC42AC2070EDFF7C67EBDF%40corpsrv2.tazznetworks.com

the important part is that we're either committing or rolling back